parquet-cli/src/main/java/org/apache/parquet/cli/BaseCommand.java

```diff
@@ -175,12 +176,13 @@ public Path qualifiedPath(String filename) throws IOException {
    * @throws IOException if there is an error creating a qualified URI
    */
   public URI qualifiedURI(String filename) throws IOException {
-    URI fileURI = URI.create(filename);
-    if (RESOURCE_URI_SCHEME.equals(fileURI.getScheme())) {
-      return fileURI;
-    } else {
-      return qualifiedPath(filename).toUri();
-    }
+    try {
+      URI fileURI = new URI(filename);
+      if (RESOURCE_URI_SCHEME.equals(fileURI.getScheme())) {
+        return fileURI;
+      }
+    } catch (URISyntaxException ignore) {}
```

Comment:
This is a behavior change that breaks the stated contract of this method: `@throws IOException if there is an error creating a qualified URI`.

In addition, the tests do not validate this change. What happens to invalid URIs after this change? Are characters automatically escaped by using `Path`? What about a resource URI with an unescaped character?

It is fine to change the method contract, but the new behavior must be as well understood as the previous behavior and it should be tested. The tests in this PR check, for example, that the file name has not changed and don't exercise the error handling code at all.

Reply:
@rdblue Thanks for your comment.

> This is a behavior change that breaks the stated contract of this method: @throws IOException if there is an error creating a qualified URI.

I don't agree. `URI.create` and `fileURI.getScheme()` don't throw `IOException` and `URISyntaxException` is not a subclass of IOException. 

- https://docs.oracle.com/javase/8/docs/api/java/net/URI.html#create-java.lang.String-
- https://docs.oracle.com/javase/8/docs/api/java/net/URISyntaxException.html

> In addition, the tests do not validate this change. What happens to invalid URIs after this change? 

The difference from `URI.create` and `new URI` is  only whether the exception to throw is the IllegalArgumentException or URISyntaxException. The reason to use `new URI` is that catching with IllegalArgumentException may catch an unintended exception.

https://hg.openjdk.java.net/jdk8u/jdk8u/jdk/file/2d4edd21bb38/src/share/classes/java/net/URI.java#l848